Feature/mx 1863 clean up asset names

erichesse opened the pull request:
### Changes

- streamline dagster asset names

erichesse commented on mex/pipeline/wikidata.py:
```diff
@@ -30,3 +32,34 @@ def extracted_organization_rki(
     return extracted_organization_rki[0]
+
+
+def get_organization_merged_id_by_query(
```
```suggestion
def get_organization_id_by_query(
```
I would prefer this name, when we're already refactoring. I find `merged` only to be confusing in this function name

rababerladuseladim replied:
Naming is based on the convenience function names for ldap in mex-common: https://github.com/robert-koch-institut/mex-common/blob/main/mex/common/ldap/extract.py

I'd rather not change all of these and I like that the function name contains the type of id explicitly, as it could also be the identifier. What do you think?

erichesse replied:
following your suggested convention the function name should then be `get_merged_id_by_query` instead

rababerladuseladim replied:
According to our discussion, it is now called `get_merged_organization_id_by_query_with_transform_and_load`

erichesse commented on mex/pipeline/wikidata.py:
```diff
@@ -30,3 +32,34 @@ def extracted_organization_rki(
+) -> dict[str, MergedOrganizationIdentifier]:
+    """Return a mapping from organizations to their stable target ID.
+
+    MergedOrganizationIdentifiers are looked up in the identity provider and will be
+    omitted for any organization that has not yet been assigned an `Identity` there.
```
I don't quite understand yet, why we are doing this for `ff_projects`, `international_projects`, `odk` and `sumo`, but not for `datscha_web`(`datscha_web_organization_ids_by_query_string`) and `grippeweb`(`grippeweb_organization_ids_by_query_string`)

rababerladuseladim replied:
I made this more straightforward to use in 7a91dd25ead0f02fd65cb9dd16740b732d240f29

This function is now also used in all extractors using wikidata and removes lots of code repetition for tranformation and loading.

erichesse reviewed (commented):
Looks really good so far. Just two issues left, I want to discuss.

erichesse commented on CHANGELOG.md:
```diff
@@ -17,6 +17,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - update mex-common to 0.26.1
 - move `mex.pipeline` documentation to `__init__` to have it in sphinx
+- move `get_organization_merged_id_by_query` to pipeline module
```
please update the changelog as well

rababerladuseladim replied:
done in ae041b4ae1ad2e0677f00d310067ae734f4636b7

erichesse reviewed (commented):
I really like the changes. Just one issue,  we transform, load and map the WikidataOrganizations, but don't extract them. So I think the module should probably rather be something like `wikidata/transform.py`

rababerladuseladim commented:
> I really like the changes. Just one issue, we transform, load and map the WikidataOrganizations, but don't extract them. So I think the module should probably rather be something like `wikidata/transform.py`

Again, I did this according to the conventions in mex-common, e.g. in https://github.com/robert-koch-institut/mex-common/blob/main/mex/common/ldap/extract.py

erichesse reviewed (approved):
Great, I really like the changes!

erichesse committed:
prefix dagster assets

erichesse committed:
merge main

erichesse committed:
fix refactoring

cutoffthetop commented on mex/extractors/biospecimen/main.py:
```diff
@@ -47,7 +47,7 @@ def biospecimen_resources() -> list[BiospecimenResource]:
 
 @asset(group_name="biospecimen")
-def extracted_mex_persons(
+def biospecimen_extracted_mex_persons(
```
why do we need to keep the `mex` in there?

```suggestion
def biospecimen_extracted_persons(
```

cutoffthetop commented on mex/extractors/synopse/main.py:
```diff
@@ -259,10 +259,10 @@ def synopse_activity() -> dict[str, Any]:
 
 @asset(group_name="synopse")
-def extracted_synopse_activities(  # noqa: PLR0913
+def synopse_activities(  # noqa: PLR0913
```
if the variable asset is called `synopse_extracted_variables` why not `synopse_extracted_activities`?

cutoffthetop reviewed (changes requested):
i have found 2 sample issues, but before i go further, i would like to settle on a naming scheme. which we should also document.

my suggestion would be for the most basic asset:

`{extractor}_{entity_type}` (eg `blueant_extracted_activity`)

reason for keeping "extracted" in there is, that we sometimes also have raw-data types that are named the same as the mex-model type. (i would expect an asset with the name `ldap_persons` to be of type `list[mex.common.ldap.models.LDAPPerson]` - whereas an asset with the name `ldap_extracted_persons` would signify a return type `list[ExtractedPerson]` extracted and transformed from LDAP.

for more complex lookup assets, i would propose:

`{extractor}_{dict value type}_by_{dict key type}` (eg `voxco_merged_organization_identifier_by_query_string`)

those assets are a bit more tricky to convert to a strict naming scheme. especially since the dict key/value types are not always very expressive (eg `synopse_str_by_str` is not very helpful). but most important for these cases would be the `{extractor}_` prefix.

erichesse committed:
round 2

erichesse commented:
> i have found 2 sample issues, but before i go further, i would like to settle on a naming scheme. which we should also document.
> 
> my suggestion would be for the most basic asset:
> 
> `{extractor}_{entity_type}` (eg `blueant_extracted_activity`)
> 
> reason for keeping "extracted" in there is, that we sometimes also have raw-data types that are named the same as the mex-model type. (i would expect an asset with the name `ldap_persons` to be of type `list[mex.common.ldap.models.LDAPPerson]` - whereas an asset with the name `ldap_extracted_persons` would signify a return type `list[ExtractedPerson]` extracted and transformed from LDAP.
> 
> for more complex lookup assets, i would propose:
> 
> `{extractor}_{dict value type}_by_{dict key type}` (eg `voxco_merged_organization_identifier_by_query_string`)
> 
> those assets are a bit more tricky to convert to a strict naming scheme. especially since the dict key/value types are not always very expressive (eg `synopse_str_by_str` is not very helpful). but most important for these cases would be the `{extractor}_` prefix.

I tried to streamline the assets to these schemes and also fixed some overlooked assets

erichesse commented:
> > i have found 2 sample issues, but before i go further, i would like to settle on a naming scheme. which we should also document.
> > my suggestion would be for the most basic asset:
> > `{extractor}_{entity_type}` (eg `blueant_extracted_activity`)
> > reason for keeping "extracted" in there is, that we sometimes also have raw-data types that are named the same as the mex-model type. (i would expect an asset with the name `ldap_persons` to be of type `list[mex.common.ldap.models.LDAPPerson]` - whereas an asset with the name `ldap_extracted_persons` would signify a return type `list[ExtractedPerson]` extracted and transformed from LDAP.
> > for more complex lookup assets, i would propose:
> > `{extractor}_{dict value type}_by_{dict key type}` (eg `voxco_merged_organization_identifier_by_query_string`)
> > those assets are a bit more tricky to convert to a strict naming scheme. especially since the dict key/value types are not always very expressive (eg `synopse_str_by_str` is not very helpful). but most important for these cases would be the `{extractor}_` prefix.
> 
> I tried to streamline the assets to these schemes and also fixed some overlooked assets

I added this to the extractor docs:
```
dagster asset naming conventions:
basic assets: {extractor}_{entity_type} (eg blueant_extracted_activity)
dict assets: {extractor}_{dict value type}_by_{dict key type} (eg voxco_merged_organization_identifier_by_query_string)
add suffixes as specifiers if asset name is not unique
```

cutoffthetop commented on mex/extractors/datenkompass/main.py:
```diff
@@ -101,7 +101,7 @@ def datenkompass_filtered_merged_organization_ids() -> set[
 
 @asset(group_name="datenkompass")
-def datenkompass_person_name_by_id() -> dict[MergedPersonIdentifier, str]:
+def datenkompass_person_str_by_id() -> dict[MergedPersonIdentifier, str]:
```
since the return type "str" is not as expressive as "...Identifier" or "Extracted..." i would have prefered the original name here. (like i wrote in the last paragraph here https://github.com/robert-koch-institut/mex-extractors/pull/83#pullrequestreview-3314055202)

but if you prefer using strictly the type (str) instead of more semantic descriptor (name), i am also ok with that

_Edit:_ or we use a combination of both, like you did in `igs_extracted_contact_points_by_mail_str` 🤔

cutoffthetop commented on mex/extractors/endnote/transform.py:
```diff
@@ -142,7 +142,7 @@ def get_doi(
     endnote_records: list[EndnoteRecord],
     endnote_bib_resource_mapping: BibliographicResourceMapping,
-    endnote_extracted_persons_by_person_string: dict[str, ExtractedPerson],
+    endnote_extracted_persons_by_person_str: dict[str, ExtractedPerson],
```
this should be either "name" or "str", right? depending on the decision for `datenkompass_person_str_by_id`

cutoffthetop commented on mex/extractors/grippeweb/main.py:
```diff
@@ -201,7 +205,7 @@ def grippeweb_extracted_parent_resource(  # noqa: PLR0913
 
 @asset(group_name="grippeweb")
-def grippeweb_extracted_variable_group(
+def grippeweb_extracted_variables_groups(
```
the plural of variable group would be "variable groups", right?

cutoffthetop reviewed (commented):
small notes/questions regarding the use of `_str_` in names. but overall i think this PR is a big step forward already in naming consistency!

erichesse committed:
update lock file

erichesse committed:
Merge branch 'main' into feature/mx_1863_clean_up_asset_names

Signed-off-by: erichesse <hessee@rki.de>

erichesse committed:
round 3

erichesse committed:
Merge branch 'feature/mx_1863_clean_up_asset_names' of https://github.com/robert-koch-institut/mex-extractors into feature/mx_1863_clean_up_asset_names

erichesse committed:
Merge branch 'main' into feature/mx_1863_clean_up_asset_names

Signed-off-by: erichesse <hessee@rki.de>

cutoffthetop reviewed (approved):
nice! thank you for your patience on this. let's merge 👍
